### ISSUE TITLE
#[cfg(unknown_option = "...") should produce a warning

### ISSUE THREAD

**Issue by Wilfred** [rust-lang/rust#39370]
The rust reference https://doc.rust-lang.org/reference.html#conditional-compilation states:

> contains a string that can be checked against (#[cfg(bar = "baz")]). Currently, only compiler-defined configuration options can have the latter form.

Given that there's a fixed list of options, could rustc warn on invalid options? The following code compiles with no warnings:

```rust
#[cfg(i_dont_exist = "foo")]
extern crate bar;

fn main() {
}
```

This caught me out today when I wrote `target = "macos"` rather than `target_os = "macos"`.

**Comment by durka** [comment]
Hmm, that line in the reference is a total lie.

```
$ rustc xx.rs --cfg 'i_dont_exist="foo"'
error[E0463]: can't find crate for `bar`
 --> xx.rs:2:1
  |
2 | extern crate bar;
  | ^^^^^^^^^^^^^^^^^ can't find crate

error: aborting due to previous error
```

**Comment by Wilfred** [comment]
Ah, sounds like I'm doomed and there's no way to warn on invalid option names. I will close this issue if there's nothing we can do to improve this.

**Comment by retep998** [comment]
There is something we can do. We can update the documentation so it no longer states "Currently, only compiler-defined configuration options can have the latter form."

**Comment by durka** [comment]
The reference should be fixed or we should add a warning, one or the other.
